debugging.rb

```diff
@@ -0,0 +1,133 @@
+----------------------------------------------------------------------------------------------------------------------------
+Debugging with puts and nil
+----------------------------------------------------------------------------------------------------------------------
+Использование puts - отличный способ отладки, но есть ОГРОМНОЕ предостережение при его использовании: вызов puts на чем-либо, что является nil 
+или пустой строкой или коллекцией, просто выведет пустую строку в ваш терминал.
+
+Это один из случаев, когда использование p приведет к большему количеству информации. Как упоминалось выше, p - это комбинация puts и #inspect, 
+последний из которых по сути печатает строковое представление того, на чем он вызван. Чтобы проиллюстрировать это, попробуйте следующее в REPL:
+
+puts "Using puts:"
+puts []
+p "Using p:"
+p []
+
+----------------------------------------------------------------------------------------------------------------------------------
+Debugging with Pry-byebug
+------------------------------------------------------------------------------------------------------------------------
+Pry - это Ruby-дополнение, которое предоставляет интерактивную оболочку REPL во время выполнения вашей программы. REPL, предоставленный Pry, очень похож 
+на IRB, но имеет дополнительные функции. Рекомендуемый Ruby-дополнение для отладки - Pry-byebug, и оно включает в себя Pry в качестве зависимости. 
+Pry-byebug добавляет отладку пошагово и навигацию по стеку.
+
+Чтобы использовать Pry-byebug, сначала вам нужно установить его в вашем терминале, выполнив команду gem install pry-byebug. Затем вы можете сделать 
+его доступным в вашей программе, добавив его в верхней части вашего файла с помощью require 'pry-byebug'. Наконец, для использования Pry-byebug вам 
+просто нужно вызвать binding.pry в любой точке вашей программы.
+
+Чтобы следовать этим примерам, сохраните код в файл Ruby (например, script.rb), а затем запустите файл в вашем терминале (например, ruby script.rb)
+	
+=end
+
+require 'pry-byebug'
+
+def isogram?(string)
+  original_length = string.length
+  string_array = string.downcase.split
+
+  binding.pry
+
+  unique_length = string_array.uniq.length
+  original_length == unique_length
+end
+
+isogram?("Odin")
+
+#----------------------------------------------------------------------------------
+
+require 'pry-byebug'
+
+def some_method
+  x = 10
+  y = 20
+  binding.pry # Остановка выполнения программы здесь
+  z = x + y
+end
+
+some_method
+
+=begin
+	
+Теперь мы будем использовать команды отладчика pry-byebug для изучения состояния программы
+
+Запустим программу и остановимся на точке останова:
+[1] pry(main)> next
+
+Посмотрим на текущее местоположение в коде:
+[2] pry(main)> whereami
+
+Просмотрим значение переменной x:
+[3] pry(main)> x
+
+Просмотрим значение переменной y:
+[4] pry(main)> y
+
+Просмотрим значение переменной z:
+[5] pry(main)> z
+
+Продолжим выполнение программы до конца:
+[6] pry(main)> continue
+	
+=end
```

Comment:
Always add a blank line to the end of file, so there is no red minus here :)

Reply:
Fixed